code/modules/mob/living/silicon/silicon_mob.dm

```diff
@@ -439,3 +474,129 @@
 	. = ..()
 	var/damage = 10 + 1.5 * speed
 	hit_atom.hit_by_thrown_mob(src, throwingdatum, damage, FALSE, FALSE)
+
+/mob/living/silicon/proc/update_hat_icons()
+	if(!silicon_hat)
+		return
+	var/image/head_icon
+
+	if(silicon_hat.icon_override)
+		hat_icon_file = silicon_hat.icon_override
+	if(!hat_icon_state)
+		hat_icon_state = silicon_hat.icon_state
+	if(isnull(hat_alpha))
+		hat_alpha = silicon_hat.alpha
+	if(!hat_color)
+		hat_color = silicon_hat.color
+
+	head_icon = get_hat_overlay()
+
+	add_overlay(head_icon)
+
+/mob/living/silicon/proc/get_hat_overlay()
+	if(!(hat_icon_file || hat_icon_state))
+		return
+	var/image/borgI = image(hat_icon_file, hat_icon_state)
+	borgI.alpha = hat_alpha
+	borgI.color = hat_color
+	borgI.pixel_y = hat_offset_y
+	if(!is_centered)
+		borgI.transform = matrix(1.125, 0, 0.5, 0, 1, 0)
+	return borgI
+
+/**
+  * Attempts to put an item on a silicon's head.
+  *
+  * Arguments:
+  * * item_to_add - The item we're attempting to place on a silicon.
+  * * user - Mob trying to put a hat on a silicon.
+  * Returns boolean reflecting if a hat was succesfully placed on the silicon.
+  */
+/mob/living/silicon/proc/place_on_head(obj/item/item_to_add, mob/user)
+	if(!item_to_add)
+		user.visible_message(
+			"<span class='notice'>[user] pats [src] on the head.</span>",
+			"<span class='notice'>You pat [src] on the head.</span>")
+		if(flags_2 & HOLOGRAM_2)
+			return FALSE
```

Comment:
You dont want to check for HOLOGRAM_2 before the visible message?

Reply:
Fair point, I've moved it above the visible message